ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/BlockHeaderValidator.java

```diff
@@ -73,7 +73,11 @@ private boolean applyRules(
     return rules
         .stream()
         .filter(filter)
-        .allMatch(rule -> rule.validate(header, parent, protocolContext));
+        .allMatch(
+            rule -> {
+              final boolean result = rule.validate(header, parent, protocolContext);
+              return result;
+            });
```

Comment:
The `result` variable appears to be redundant here.

Reply:
this shouldn't have gone in explicitly - however, it does make it easier to determine which rule has failed during debug (is it actually required, or is there an easy way to debug the allMatch?)